modules/shortcodes/instagram.php

```diff
@@ -262,6 +262,28 @@ function jetpack_shortcode_instagram( $atts ) {
 		return '';
 	}
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		$url_pattern = '#http(s?)://(www\.)?instagr(\.am|am\.com)/p/([^/?]+)#i';
```

Comment:
This logic is mainly [taken from the AMP plugin](https://github.com/ampproject/amp-wp/blob/develop/includes/embeds/class-amp-instagram-embed.php#L67).